File: tests/integration/octave_modular/regression.m
function y = regression(filename)
	init_shogun;
	y=true;
	addpath('util');
	addpath('../data/regression');

	eval('globals'); % ugly hack to have vars from filename as globals
	eval(filename);
	prefix='regression_';

	if ~set_features('kernel_')
		return;
	end
	if ~set_kernel()
		return;
	end
	kernel.parallel.set_num_threads(regression_num_threads);

	lab=RegressionLabels(regression_labels);

	if strcmp(regression_name, 'KERNELRIDGEREGRESSION')==1
		regression=KernelRidgeRegression(regression_tau, kernel, lab);

	elseif strcmp(regression_name, 'LibSVR')==1
		regression=LibSVR(regression_C, regression_epsilon, kernel, lab);
		regression.set_tube_epsilon(regression_tube_epsilon);

	elseif strcmp(regression_name, 'SVRLight')==1
		try
			regression=SVRLight(regression_C, regression_epsilon, kernel, lab);
			regression.set_tube_epsilon(regression_tube_epsilon);
		catch
			disp('No support for SVRLight available.');
			return;
		end

	else
		error('Unsupported regression %s!', regression_name);
	end

	regression.parallel.set_num_threads(regression_num_threads);
	regression.train();

	bias=0;
	if ~isempty(regression_bias)
		bias=regression.get_bias();
		bias=abs(bias-regression_bias);
	end

	alphas=0;
	sv=0;
	if ~isempty(regression_alpha_sum) && ~isempty(regression_sv_sum)
			tmp=regression.get_alphas();
			for i = 1:length(tmp)
				alphas=alphas+tmp(i:i);
			end
			alphas=abs(alphas-regression_alpha_sum);
			tmp=regression.get_support_vectors();
			for i = 1:length(tmp)
				sv=sv+tmp(i:i);
			end
			sv=abs(sv-regression_sv_sum);
	end

	kernel.init(feats_train, feats_test);
	classified=max(abs(
		regression.apply().get_labels()-regression_classified));

	data={'classifier', alphas, bias, sv, classified};
	y=check_accuracy(regression_accuracy, data);

File: tests/integration/r_modular/regression.R
regression <- function(filename) {
	source('util/get_features.R')
	source('util/get_kernel.R')
	source('util/check_accuracy.R')
	source('util/tobool.R')

	feats <- get_features('kernel_')
	if (typeof(feats)=='logical') {
		return(TRUE)
	}
	kernel <- get_kernel(feats)
	if (typeof(kernel)=='logical') {
		return(TRUE)
	}

	regression_num_threads <- as.integer(regression_num_threads)
	kernel$parallel$set_num_threads(kernel$parallel, regression_num_threads)
	lab <- Labels(as.double(regression_labels))

	if (regexpr('KRR', regression_name)>0) {
		regression <- KRR(regression_tau, kernel, lab)
	} else if (regexpr('LibSVR', regression_name)>0) {
		regression <- LibSVR(regression_C, regression_epsilon, kernel, lab)
		regression$set_tube_epsilon(regression, regression_tube_epsilon)
	} else if (regexpr('SVRLight', regression_name)) {
		try(regression <- SVRLight(
			regression_C, regression_epsilon, kernel, lab))
		regression$set_tube_epsilon(regression, regression_tube_epsilon)
	} else {
		print(paste('Unsupported regression', regression_name))
		return(FALSE)
	}

	regression$parallel$set_num_threads(
		regression$parallel, regression_num_threads)
	regression$train(regression)

	bias <- 0
	if (exists('regression_bias')) {
		bias <- abs(regression$get_bias()-regression_bias)
	}

	alphas <- 0
	sv <- 0
	if (exists('regression_alpha_sum')) {
		tmp <- regression$get_alphas()
		for (i in 1:length(tmp) ){
			alphas <= alphas + tmp[i]
		}
		alphas <- abs(alphas-regression_alpha_sum)

		tmp <- regression$get_support_vectors()
		for (i in 1:length(tmp)) {
			sv <- sv + tmp[i]
		}
		sv <- abs(sv-regression_sv_sum)
	}

	kernel$init(kernel, feats[[1]], feats[[2]])
	classified <- regression$classify(regression)
	classified <- max(abs(
		classified$get_labels(classified)-regression_classified))

	data <- list(alphas, bias, sv, classified)
	return(check_accuracy(regression_accuracy, 'classifier', data))
}
